### PR TITLE
Revert isCacheableValue to deny null values. Remove null value tests …

### PR DESCRIPTION
**Pull request by jeff-kilbride**
…and add a test for the .wrap method.

See #56.

### CODE REVIEW

**Comment by jeff-kilbride** [comment]
Just a reminder...  this PR is to `master`.

It's very late here, so I am done for the night. I'll check in tomorrow.

File: test/lib/redis-store-spec.js [hunk 1]
```diff
@@ -84,6 +85,7 @@ describe('set', function () {
     });
   });
 
+  /*
   it('should store a null value without error', function (done) {
```
Comment (mrister) [inline test/lib/redis-store-spec.js]: We  shouldn't really leave commented the code and it is good to have null test (in case of the change to prove that null is not cached in the store) if there is not another one (can't really see from github PR)

File: test/lib/redis-store-spec.js [hunk 2]
```diff
@@ -336,7 +339,7 @@ describe('isCacheableValue', function () {
     assert.equal(redisCache.store.isCacheableValue(100), true);
     assert.equal(redisCache.store.isCacheableValue(''), true);
     assert.equal(redisCache.store.isCacheableValue('test'), true);
-    assert.equal(redisCache.store.isCacheableValue(null), true);
+    //assert.equal(redisCache.store.isCacheableValue(null), true);
```
Comment (mrister) [inline test/lib/redis-store-spec.js]: same here either remove or assert expectancy to be change to false.

File: test/lib/redis-store-spec.js [hunk 3]
```diff
@@ -463,3 +466,39 @@ describe('defaults', function () {
...
+    setTimeout(function () {
+      cb(null, {id: id, name: 'Bob'});
+    }, 100);
+  }
+
+  var userId = 123;
+  var key = 'user_' + userId;
+  var fromCache = true;
+
+  it('should work with wrap function', function (done) {
```
Comment (mrister) [inline test/lib/redis-store-spec.js]: I am not sure this test is actually testing anything relevant. It should at least use the getUser to get something from a (redis store.

File: test/lib/redis-store-spec.js [hunk 4]
```diff
@@ -463,3 +466,39 @@ describe('defaults', function () {
...
+      assert.equal(user.name, 'Bob');
+
+      // Second call to wrap should retrieve from cache
+      fromCache = true;
+      redisCache.wrap(key, function (cb) {
+        fromCache = false;
+        getUser(userId, cb);
+      }, function (err, user) {
+        assert.equal(user.name, 'Bob');
+        assert.equal(fromCache, true);
```
Comment (mrister) [inline test/lib/redis-store-spec.js]: it is true because you set it to true on line 493 (or 480) and not because it is cached.

File: test/lib/redis-store-spec.js [hunk 5]
```diff
@@ -463,3 +466,39 @@ describe('defaults', function () {
   });
 });
 
+describe('wrap function', function () {
```
Comment (mrister) [inline test/lib/redis-store-spec.js]: we should add some more tests for nonexistent keys (null, undefined) that we know are causing people issues #57

File: test/lib/redis-store-spec.js (hunk 3 above)
Comment (PuKoren) [inline test/lib/redis-store-spec.js]: From what I seen it does: he tests that the callback function is executed and that the returned object is the one inserted, and then that the callback function is NOT executed and that the returned object is the one previously inserted (aka stored in store)

Comment (mrister) [inline test/lib/redis-store-spec.js]: Oh ok that makes sense. I was just tricked by scope reach of the functions and point of the test. We should test for some of our cases with redis(store) and not if wrap is working or not as it is not the function we provide. I'll remove the comment related to this test.

**Comment by PuKoren** [comment]
@mrister applied your review on this branch while @jeff-kilbride is off https://github.com/dial-once/node-cache-manager-redis/compare/master...hotfix/0.2.6

**Comment by mrister** [comment]
@PuKoren looks good!

**Comment by PuKoren** [comment]
Ok, once this PR is accepted, I propose that we close it and apply the hotfix on master/develop with tags, following our git flow guidelines, and then we can discard this PR.

I cherry-picked the commit of @jeff-kilbride on the hotfix branch so his contribution is kept in history, just saying

**Comment by mrister** [comment]
Then I approve :-) 💯

**Comment by PuKoren** [comment]
Done, released in 0.3.0 and backported in 0.2.6
